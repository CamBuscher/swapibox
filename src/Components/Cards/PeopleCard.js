import React from 'react';
import PropTypes from 'prop-types';

const PeopleCard = ({person, toggleFavorite, isFavorite}) => {
  const favoriteIcon = isFavorite ?
    <img src='/images/fullStar.png' className='favoriteIcon favorite' title='Click to remove from favorites' onClick={() => toggleFavorite(person)} />
    :
    <img src='/images/star.png' className='favoriteIcon notFavorite' title='Click to add to favorites' onClick={() => toggleFavorite(person)} />;

  return (
    <div className='card peopleCard'>
      <h3>{person.name}</h3>
      <h4>{person.species}</h4>
      {favoriteIcon}
      <p>Homeworld: {person.homeworld}</p>
      <p>Population: {person.homeworldPop}</p>
    </div>
  );
};

PeopleCard.propTypes = {
  person: PropTypes.object.isRequired,
  toggleFavorite: PropTypes.func.isRequired,
  isFavorite: PropTypes.bool.isRequired
};

export default PeopleCard;