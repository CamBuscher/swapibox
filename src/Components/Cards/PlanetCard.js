import React from 'react';
import PropTypes from 'prop-types';

const PlanetCard = ({planet, toggleFavorite, isFavorite}) => {
  const favoriteIcon = isFavorite ?
    <img src='/images/fullStar.png' className='favoriteIcon' title='Click to remove from favoritesClick to add to favorites' onClick={() => toggleFavorite(planet)} /> 
    :
    <img src='/images/star.png' className='favoriteIcon' title='Click to add to favorites' onClick={() => toggleFavorite(planet)} />;

  const residents = () => {
    if (planet.residents.length) {
      return <div>
        <p>Notable residents:</p>
        <ul>
          {planet.residents.map((resident, index) => <li key={`${index} resident`}>{resident}</li>)};
        </ul>
      </div>;
    } else {
      return <p> No notable resients </p>;
    }
  };
  return <div className="card planetsCard">
    <h3>{planet.name}</h3>
    <h4>Population: {planet.population}</h4>
    {favoriteIcon}
    <p>Climate: {planet.climate}</p>
    <p>Terrain: {planet.terrain}</p>
    {residents()}
  </div>;
};

PlanetCard.propTypes = {
  planet: PropTypes.object.isRequired,
  toggleFavorite: PropTypes.func.isRequired,
  isFavorite: PropTypes.bool.isRequired
};

export default PlanetCard;