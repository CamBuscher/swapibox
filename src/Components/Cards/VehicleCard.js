import React from 'react';
import PropTypes from 'prop-types';

const VehicleCard = ({vehicle, toggleFavorite, isFavorite}) => {
  const favoriteIcon = isFavorite ?
    <img src='/images/fullStar.png' className='favoriteIcon favorite' title='Click to remove from favorites' onClick={() => toggleFavorite(vehicle)} />
    :
    <img src='/images/star.png' className='favoriteIcon notFavorite' title='Click to add to favorites' onClick={() => toggleFavorite(vehicle)} />;

  return (
    <div className="card vehiclesCard">
      <h3>{vehicle.name}</h3>
      <h5>Model: {vehicle.model}</h5>
      {favoriteIcon}
      <p>Number of passengers: {vehicle.numPassengers}</p>
      <p>Class: {vehicle.class} </p>
    </div>
  );
};

VehicleCard.propTypes = {
  vehicle: PropTypes.object.isRequired,
  toggleFavorite: PropTypes.func.isRequired,
  isFavorite: PropTypes.bool.isRequired
};

export default VehicleCard;