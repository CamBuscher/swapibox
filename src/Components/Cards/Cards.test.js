import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import PeopleCard from './PeopleCard';
import PlanetCard from './PlanetCard';
import VehicleCard from './VehicleCard';


describe('Card components', () => {
  describe('PeopleCard', () => {
    let wrapper;
    let mockProps;
    
    beforeEach(() => {
      mockProps = {
        person: {name: 'onyx'},
        toggleFavorite: jest.fn(),
        isFavorite: false
      }
      wrapper = mount(<PeopleCard {...mockProps}/>)
    });

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have an icon with class of favorite if it is a favorite', () => {
      wrapper = shallow(<PeopleCard {...mockProps} isFavorite={true} />);

      const icon = wrapper.find('div').find('img')
      expect(icon.hasClass('favorite')).toBe(true);
    });

    it('should have an icon with class of notFavorite if it is not a favorite', () => {
      wrapper = shallow(<PeopleCard {...mockProps}/>);

      const icon = wrapper.find('div').find('img')
      expect(icon.hasClass('notFavorite')).toBe(true);
    });

    it('should call toggleFavorite with person as a param when icon is clicked', () => {
      const icon = wrapper.find('div').find('img');
      icon.simulate('click');

      expect(wrapper.prop('toggleFavorite')).toHaveBeenCalledWith({name: 'onyx'});
    });
  });

  describe('VehicleCard', () => {
    let wrapper;
    let mockProps;

    beforeEach(() => {
      mockProps = {
        vehicle: { name: 'prius' },
        toggleFavorite: jest.fn(),
        isFavorite: false
      };
      wrapper = mount(<VehicleCard {...mockProps} />);
    });

    it('matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have an icon with class of favorite if it is a favorite', () => {
      wrapper = shallow(<VehicleCard {...mockProps} isFavorite={true} />);

      const icon = wrapper.find('div').find('img')
      expect(icon.hasClass('favorite')).toBe(true);
    });

    it('should have an icon with class of notFavorite if it is not a favorite', () => {
      wrapper = shallow(<VehicleCard {...mockProps} />);

      const icon = wrapper.find('div').find('img');
      expect(icon.hasClass('notFavorite')).toBe(true);
    });

    it('should call toggleFavorite with person as a param when icon is clicked', () => {
      const icon = wrapper.find('div').find('img');
      icon.simulate('click');

      expect(wrapper.prop('toggleFavorite')).toHaveBeenCalledWith({ name: 'prius' });
    });
  })
});