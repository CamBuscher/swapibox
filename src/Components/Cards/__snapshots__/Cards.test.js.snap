// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card components PeopleCard matches snapshot 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <PeopleCard
    isFavorite={false}
    person={
        Object {
            "name": "onyx",
          }
    }
    toggleFavorite={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "isFavorite": false,
      "person": Object {
        "name": "onyx",
      },
      "toggleFavorite": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="card peopleCard"
>
        <h3>
                onyx
        </h3>
        <h4 />
        <img
                src="/images/star.png"
                class="favoriteIcon notFavorite"
                title="Click to add to favorites"
        />
        <p>
                Homeworld: 
        </p>
        <p>
                Population: 
        </p>
</div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            onyx
</h3>,
          <h4 />,
          <img
            className="favoriteIcon notFavorite"
            onClick={[Function]}
            src="/images/star.png"
            title="Click to add to favorites"
/>,
          <p>
            Homeworld: 
</p>,
          <p>
            Population: 
</p>,
        ],
        "className": "card peopleCard",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <h3>
            onyx
</h3>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "onyx",
          },
          "ref": null,
          "rendered": Array [
            "onyx",
          ],
          "type": "h3",
        },
        Object {
          "instance": <h4 />,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "h4",
        },
        Object {
          "instance": <img
            src="/images/star.png"
            class="favoriteIcon notFavorite"
            title="Click to add to favorites"
/>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "favoriteIcon notFavorite",
            "onClick": [Function],
            "src": "/images/star.png",
            "title": "Click to add to favorites",
          },
          "ref": null,
          "rendered": Array [
            undefined,
          ],
          "type": "img",
        },
        Object {
          "instance": <p>
            Homeworld: 
</p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
          ],
          "type": "p",
        },
        Object {
          "instance": <p>
            Population: 
</p>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Population: ",
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Population: ",
          ],
          "type": "p",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "isFavorite": false,
        "person": Object {
          "name": "onyx",
        },
        "toggleFavorite": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="card peopleCard"
>
          <h3>
                    onyx
          </h3>
          <h4 />
          <img
                    src="/images/star.png"
                    class="favoriteIcon notFavorite"
                    title="Click to add to favorites"
          />
          <p>
                    Homeworld: 
          </p>
          <p>
                    Population: 
          </p>
</div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              onyx
</h3>,
            <h4 />,
            <img
              className="favoriteIcon notFavorite"
              onClick={[Function]}
              src="/images/star.png"
              title="Click to add to favorites"
/>,
            <p>
              Homeworld: 
</p>,
            <p>
              Population: 
</p>,
          ],
          "className": "card peopleCard",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <h3>
              onyx
</h3>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "onyx",
            },
            "ref": null,
            "rendered": Array [
              "onyx",
            ],
            "type": "h3",
          },
          Object {
            "instance": <h4 />,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "h4",
          },
          Object {
            "instance": <img
              src="/images/star.png"
              class="favoriteIcon notFavorite"
              title="Click to add to favorites"
/>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "favoriteIcon notFavorite",
              "onClick": [Function],
              "src": "/images/star.png",
              "title": "Click to add to favorites",
            },
            "ref": null,
            "rendered": Array [
              undefined,
            ],
            "type": "img",
          },
          Object {
            "instance": <p>
              Homeworld: 
</p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Homeworld: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Homeworld: ",
            ],
            "type": "p",
          },
          Object {
            "instance": <p>
              Population: 
</p>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Population: ",
                undefined,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Population: ",
            ],
            "type": "p",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
